crates/types/src/lib.rs

```diff
@@ -120,39 +120,41 @@ macro_rules! impl_option_non_zero {
 
 impl_option_non_zero!(u8, u16, u32, u64, usize);
 
-impl<P> ValueEncoder<General> for OptionNonZero<P>
+impl<const _P: u8, P> ValueEncoder<GeneralGeneric<_P>, OptionNonZero<P>> for ()
```

Comment:
I was actually wondering if we can -later- drop this completely and implement a custom encoder for `NonZero` and probably other std and external types like `RangeInclusive` for example. I think the new changes should make it easier and then we can get rid of those noisy proxy types? What do you think?

Reply:
> NonZero* types can definitely be encoded and the most straightforward way to do it would be pretty simple (they would have no "empty" state). Having a wrapper type that re-grants them value-encoding by encoding None as the zero value is still a separate exercise i think

that's exactly how I was planning to do it :) it also works fine in my playground repo